src/golang/cmd/server/handler/refresh_workflow.go

```diff
@@ -79,44 +103,63 @@ func (h *RefreshWorkflowHandler) Prepare(r *http.Request) (interface{}, int, err
 	}, http.StatusOK, nil
 }
 
-func generateWorkflowJobName() string {
-	return fmt.Sprintf("workflow-adhoc-%s", uuid.New().String())
-}
-
 func (h *RefreshWorkflowHandler) Perform(ctx context.Context, interfaceArgs interface{}) (interface{}, int, error) {
 	args := interfaceArgs.(*RefreshWorkflowArgs)
 
-	workflowObject, err := h.WorkflowReader.GetWorkflow(
+	dag, err := utils.ReadLatestWorkflowDagFromDatabase(
 		ctx,
 		args.WorkflowId,
+		h.WorkflowReader,
+		h.WorkflowDagReader,
+		h.OperatorReader,
+		h.ArtifactReader,
+		h.WorkflowDagEdgeReader,
 		h.Database,
 	)
 	if err != nil {
-		if err == database.ErrNoRows {
-			return nil, http.StatusBadRequest, errors.New("Unable to find workflow.")
-		}
-		return nil, http.StatusInternalServerError, errors.New("Unable to find workflow.")
+		return nil, http.StatusInternalServerError, errors.Wrap(err, "Error reading dag object from database.")
 	}
 
-	jobName := generateWorkflowJobName()
-
-	jobSpec := job.NewWorkflowSpec(
-		workflowObject.Name,
-		workflowObject.Id.String(),
-		h.Database.Config(),
-		h.Vault.Config(),
-		h.JobManager.Config(),
-		h.GithubManager.Config(),
-		args.Parameters,
+	workflowDag, err := dag_utils.NewWorkflowDag(
```

Comment:
Right now we aren't using the `args.Parameters` that are passed in. We'll need to add the user-defined parameters to the new `dbWorkflowDag` that we pass in here. Something like this:
```
// Overwrite the "default" values in the operator spec for this workflowDag.
	// Because this workflowDag is never written to the database, we will not contaminate
	// the default in the db.
	if ex.Parameters != nil {
		for name, newVal := range ex.Parameters {
			op := dbWorkflowDag.GetOperatorByName(name)
			if op == nil {
				continue
			}

			if !op.Spec.IsParam() {
				return errors.Newf("Cannot set parameters on a non-parameter operator %s", name)
			}
			dbWorkflowDag.Operators[op.Id].Spec.Param().Val = newVal
		}
	}
```
from https://github.com/aqueducthq/aqueduct/blob/main/src/golang/cmd/executor/executor/workflow.go#L89 should help you do this.

Reply:
Have Updated this change!